fe/fe-core/src/main/java/com/starrocks/credential/AWSCloudCredential.java

```diff
@@ -0,0 +1,225 @@
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.auth.DefaultAWSCredentialsProviderChain;
+import com.amazonaws.auth.InstanceProfileCredentialsProvider;
+import com.amazonaws.auth.STSAssumeRoleSessionCredentialsProvider;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenService;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClientBuilder;
+import com.google.common.base.Preconditions;
+import com.starrocks.credential.provider.AssumedRoleCredentialProvider;
+import com.starrocks.thrift.TCloudProperty;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.s3a.AnonymousAWSCredentialsProvider;
+
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * Authenticating process (It's a pseudocode code):
+ * Credentials credentials = null;
+ * if (useAWSSDKDefaultBehavior) {
+ *     return new DefaultAWSCredentialsProviderChain();
+ * } else if (useInstanceProfile) {
+ *      credentials = GetInstanceProfileCredentials();
+ *     if (useIamRoleArn) {
+ *        credentials = GetAssumeRole(credentials, iamRoleArn, externalId);
+ *     }
+ *     return credentials;
+ * } else if (exist(accessKey) && exist(secretKey)) {
+ *     credentials = GetAKSKCredentials(accessKey, secretKey);
+ *     if (useIamRoleArn) {
+ *         credentials = GetAssumeRole(credentials, iamRoleArn, externalId);
+ *     }
+ *     return credentials;
+ * } else {
+ *     // Unreachable!!!!
+ *     // We don't allowed to create anonymous credentials, we will check it in validate() method.
+ *     // If user want to use anonymous credentials, they just don't set cloud credential directly.
+ * }
+ */
+public class AWSCloudCredential implements CloudCredential {
+
+    private final boolean useAWSSDKDefaultBehavior;
+
+    private final boolean useInstanceProfile;
+
+    private final String accessKey;
+
+    private final String secretKey;
+
+    private final String iamRoleArn;
+
+    private final String externalId;
+
+    private final String region;
+
+    private final String endpoint;
+
+    protected AWSCloudCredential(boolean useAWSSDKDefaultBehavior, boolean useInstanceProfile, String accessKey,
+                              String secretKey, String iamRoleArn, String externalId, String region, String endpoint) {
+        Preconditions.checkNotNull(accessKey);
+        Preconditions.checkNotNull(secretKey);
+        Preconditions.checkNotNull(iamRoleArn);
+        Preconditions.checkNotNull(externalId);
+        Preconditions.checkNotNull(region);
+        Preconditions.checkNotNull(endpoint);
+        this.useAWSSDKDefaultBehavior = useAWSSDKDefaultBehavior;
+        this.useInstanceProfile = useInstanceProfile;
+        this.accessKey = accessKey;
+        this.secretKey = secretKey;
+        this.iamRoleArn = iamRoleArn;
+        this.externalId = externalId;
+        this.region = region;
```

Comment:
better to move `region` and `endpoint` to `CloudConfiguration`

Reply:
We should consider it for `create credential` for the future